lib/transport/transport-tls.c

```diff
@@ -34,15 +34,66 @@ typedef struct _LogTransportTLS
 {
   LogTransportSocket super;
   TLSSession *tls_session;
+  gboolean sending_shutdown;
 } LogTransportTLS;
 
+static inline gboolean
+_is_shutdown_sent(gint shutdown_rc)
+{
+  return shutdown_rc >= 0;
+}
+
+static gint
+log_transport_tls_send_shutdown(LogTransportTLS *self)
+{
+  self->sending_shutdown = TRUE;
+  gint shutdown_rc = SSL_shutdown(self->tls_session->ssl);
+
+  if (_is_shutdown_sent(shutdown_rc))
+    {
+      self->sending_shutdown = FALSE;
+      return shutdown_rc;
+    }
+
+  gint ssl_error = SSL_get_error(self->tls_session->ssl, shutdown_rc);
+
+  switch (ssl_error)
```

Comment:
I'd factor out this part of the function (as this part does not modify the return value, it would be more readable).

Reply:
Extracted, thanks.
The method name might not be the best.